libs/domains/service-terraform/feature/src/lib/terraform-configuration-settings/terraform-configuration-settings.tsx

```diff
@@ -114,18 +116,31 @@ export const TerraformConfigurationSettings = ({
           <Controller
             name="provider_version.explicit_version"
             control={methods.control}
-            render={({ field }) => (
-              <InputSelect
-                label="Terraform version"
-                value={field.value}
-                isLoading={isTerraformVersionLoading}
-                onChange={field.onChange}
-                options={versions
-                  .filter((v) => v.engine === methods.watch('engine'))
-                  .map((v) => ({ label: v.version, value: v.version }))}
-                hint="Select the Terraform version to use for the service"
-              />
-            )}
+            render={({ field }) => {
+              const versionOptions = versions
+                .filter((v) => v.engine === methods.watch('engine'))
+                .map((v) => ({ label: v.version, value: v.version }))
+
+              // Pick the same selected version or default to the latest one available in the list
+              const selectedValue = versionOptions.find((option) => option.value === selectedVersionRef.current)
+                ? selectedVersionRef.current
+                : versionOptions[0]?.value || ''
+              selectedVersionRef.current = selectedValue
```

Comment:
@erebe It is not optimal to have this logic here given this function will be executed every time the component re-renders. Instead these variables should be wrapped with a `useMemo` and declared in the component's main function 🙏

Reply:
Do you think this function is really necessary? It look complex for simple case and we don't manage it in the product

Reply:
ok ok, let's go without it then. In any case this component must be change because it is not the same between the creation flow and the setting page